[ENHANCEMENT] Week 7 Cutscenes no longer a video (feat. @CrusherNotDrip)

TechnikTil opened the pull request:
<!-- Please read the Contributing Guide (https://github.com/FunkinCrew/Funkin/blob/main/docs/CONTRIBUTING.md) before submitting this PR. -->
### NOTE: Supersedes #32

<!-- Briefly describe the issue(s) fixed. -->
# What is this pull request?
This Pull Request redoes the Week 7 cutscenes but everything happens in-game instead of as a video.

Here is a video of them (It may look like the mp4 but even poorer quality (had to compress it) but it shows the countdown transition so u can still kinda see what its like in-game):

https://github.com/user-attachments/assets/29683fa0-b3a4-45c3-a08b-38436dad838f


<!-- Does this PR close any issues? If so, link them below. -->
## Linked Issues
FunkinCrew/Funkin#3324

CrusherNotDrip commented:
@TechnikTil I don't have the ability to change stuff

AbnormalPoof reviewed (changes requested):
Not a bad start! Some notes:

- Camera movements aren't completely accurate, you can reference legacy's code for more accurate camera movements: https://github.com/FunkinCrew/Funkin/tree/legacy/0.2.x
- Dave says the music should fade out instead of abruptly stopping.
- Dave says BF should not bop to the beat. This likely applies to the stage props as well.
- For GF turning demon, you can quickly swap out the atlas for a sparrow, as that animation uses a blend mode that currently doesn't work properly in-game.

CrusherNotDrip commented:
Wait a second we have a pr open that requires some changes![image](https://github.com/user-attachments/assets/a0c14b41-d1c8-451b-982a-801cf82c1385)

CrusherNotDrip commented:
> blah blah blah crusher it no embed

yeaim well aware

> Message ID: ***@***.***>
>

TechnikTil commented:
i really gotta squash this sometime soon

TechnikTil commented:
also ill keep this as a draft for now

TechnikTil commented:
LETSS GOOOO

TechnikTil committed:
Put Week 7 Cutscenes in-game instead of being a video.

Commit Names:
- Export Tankman and Pico (GF) atlases with Better TA
- Ugh & Guns progress made.
- goddamn it
- Stress Cutscene, and Pico re-export.
- TechnikStupid
- this dude only fixed it for stress
- Clean Up
- tiny changes to keep dave happy (and others)
- ugh cutscene skipping
- Re-export atlases.
- Guns + Stress skipping and tiny patch ups
- More accurate camera positions and tiny fixes
- Subtitles (Work in Progress)
- final touch ups, getting ready to squash
- shit shit shit shit shit shit shit
- destroy subtitles early (since u can see them above `camCutscene`), also run formatter
- use stage character position instead of direct character position, also re-export gf
- get some stuff ready for mr. squash

Co-authored-by: CrusherNotDrip <crushernotdrip@tilnotdrip.org>
Signed-off-by: TechnikTil <techniktil@tilnotdrip.org>

Raltyro reviewed (approved):
This looks good? Hoping to see this merged by the next update or hotfix

Raltyro reviewed (changes requested):
The only complains i have is that the camera are too fast to transition from cutscenes to gameplay, but anything else its good

https://github.com/user-attachments/assets/c1d7e5f5-b436-49a4-acb1-ba36e9178148

Hundrec commented:
Leaving a note here to consider the detail in https://github.com/FunkinCrew/Funkin/issues/7394

7oltan commented:
chop chop buddy great sort this pr NOW!

CrusherNotDrip commented:
unchop unchop hot reloading sum BUNS last time we tried :sob:
